pkg/operator/ceph/object/notification/controller.go

```diff
@@ -237,26 +197,46 @@ func (r *ReconcileNotifications) updateCephBucketNotification(notification *ceph
 			return reconcile.Result{}, err
 		}
 
-		// provision the notification
-		provisioner := Provisioner{
-			Client:           r.client,
-			Context:          r.context,
-			ClusterInfo:      r.clusterInfo,
-			ClusterSpec:      r.clusterSpec,
-			opManagerContext: r.opManagerContext,
-		}
-		session, err := provisioner.createSession(
-			ob.Spec.AdditionalState["cephUser"],
-			objectStoreName,
-		)
+		s3Agent, err := NewS3Agent(r.opManagerContext, r.context, clusterInfo, &cephCluster.Spec, ob.Spec.AdditionalState["cephUser"], objectStoreName)
 		if err != nil {
-			return reconcile.Result{}, errors.Wrapf(err, "failed to create session for CephBucketNotification provisioning %q for ObjectBucketClaims %q", bnName, bucketName)
+			return reconcile.Result{}, errors.Wrapf(err, "failed to create S3 agent for CephBucketNotification provisioning %q for ObjectBucketClaims %q", bnName, bucketName)
 		}
-		err = provisioner.Create(&ob, topicARN, notification, session)
+		err = Create(&ob, topicARN, notification, s3Agent)
 		if err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "failed to provision CephBucketNotification %q for ObjectBucketClaims %q", bnName, bucketName)
 		}
 	}
 
 	return reconcile.Result{}, nil
 }
+
+func getCephObjectStoreName(ob bktv1alpha1.ObjectBucket) (objectStoreName types.NamespacedName, err error) {
```

Comment:
```suggestion
func getCephObjectStoreName(ob bktv1alpha1.ObjectBucket) (types.NamespacedName, error) {
```

While we are at it, can we stop naming returns? We very rarely do it in our code base and prefer returning like


```go
return objectStoreName, nil
```

Reply:
sure, can change that.
as i'm not "native" in golang :-) is there a general guideline to avoid that?
or is this just a specific style n rook?

Reply:
There is no general guidelines to avoid this, and actually, effective go "recommends" it or at least says it can be useful for clarity: https://golang.org/doc/effective_go#named-results. However, I haven't seen many projects doing it and Rook certainly does not. I believe we have 1 exception that works with a `defer` statement and that's it.

Reply:
ok. will follow the "When in Rome.." guideline

Reply:
i cleaned that up in other places in the code. kepts it in one place, where it actually add clarity. I have a function:
```golang
func getUserCredentials(...) (accessKey string, secretKey string, err error)
```
where naming the return strings allows the caller to understand what are the return values without looking into the implementation of the function